siteapp/views.py

```diff
@@ -1370,15 +1370,24 @@ def move_project(request, project_id):
     request ([HttpRequest]): The network request
     project_id ([int|str]): The id of the project
     Returns:
-        [JsonResponse]: Either a ok status or an error
+        [JsonResponse]: Either an ok status or an error
     """
-    try:
-        new_portfolio_id = request.POST.get("new_portfolio", "").strip() or None
+      try:
+        new_portfolio_id = request.POST.get("new_portfolio", "").strip()
         project = get_object_or_404(Project, id=int(project_id))
         cur_portfolio = project.portfolio
         new_portfolio = get_object_or_404(Portfolio, id=int(new_portfolio_id))
+    except:
+        return JsonResponse({"status": "error", "message": "Portfolio entered does not exist."})
+
+    # Check if the user moving the project is a superuser or
+    # if they are the owner of the project and have edit permissions in the target directory
+    if request.user.is_superuser or ((request.user in project.get_admins()) and 'change_portfolio' in get_perms(request.user, new_portfolio)):
```

Comment:
```suggestion
    owner = True if request.user.has_perm('can_grant_portfolio_owner_permission', cur_portfolio) else False
    if request.user.is_superuser or ((request.user in project.get_admins()) or owner or 'change_portfolio' in get_perms(request.user, new_portfolio)):
```
@ztzerhouni 
You already check edit access with the change_portfolio and admin status with project.get_admins but you also need to check if they are an owner.
- [X] 1. Checks if the user is a superuser and can move the portfolio regardless of any other details.
- [ ] 2. Checks if the user is the owner/admin of the project and has edit access to the target portfolio.
To check if the user is an owner of the target portfolio this is an option.
`owner = True if request.user.has_perm('can_grant_portfolio_owner_permission', new_portfolio) else False`
- [X] 3. Users continue to have access to a project after it moves. It is accessible from the project tab in the nav bar.
- [X] 4. Each user who is a member of the project is given view access to the target portfolio.